Index: pcsx2.snapshot-3369/common/src/Utilities/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3369.orig/common/src/Utilities/CMakeLists.txt
+++ pcsx2.snapshot-3369/common/src/Utilities/CMakeLists.txt
@@ -46,13 +46,6 @@
 
 
 set(BadFlags_O2
-    -fno-expensive-optimizations
-    -fno-forward-propagate
-    -fno-inline-small-functions
-    -fno-ipa-cp
-    -fno-schedule-insns2
-    -fno-strict-aliasing
-    -fno-tree-builtin-call-dce
     )
 
 
Index: pcsx2.snapshot-3369/common/src/x86emitter/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3369.orig/common/src/x86emitter/CMakeLists.txt
+++ pcsx2.snapshot-3369/common/src/x86emitter/CMakeLists.txt
@@ -46,13 +46,6 @@
 
 
 set(BadFlags_O2
-    -fno-expensive-optimizations
-    -fno-forward-propagate
-    -fno-inline-small-functions
-    -fno-ipa-cp
-    -fno-schedule-insns2
-    -fno-strict-aliasing
-    -fno-tree-builtin-call-dce
     )
 
 # set optimization flags
Index: pcsx2.snapshot-3369/pcsx2/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3369.orig/pcsx2/CMakeLists.txt
+++ pcsx2.snapshot-3369/pcsx2/CMakeLists.txt
@@ -42,13 +42,6 @@
     )
 
 set(BadFlags_O2
-    -fno-expensive-optimizations
-    -fno-forward-propagate
-    -fno-inline-small-functions
-    -fno-ipa-cp
-    -fno-schedule-insns2
-    -fno-strict-aliasing
-    -fno-tree-builtin-call-dce
     )
 
 # set optimization flags
