Index: pcsx2.snapshot-3369/common/src/Utilities/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3369.orig/common/src/Utilities/CMakeLists.txt
+++ pcsx2.snapshot-3369/common/src/Utilities/CMakeLists.txt
@@ -39,21 +39,9 @@
 
 # Remove optimization that can break the code. Must be retested
 set(BadFlags_O0
-    -fno-guess-branch-probability
-    -fno-dse
-    -fno-tree-dse
     )
 
 set(BadFlags_O1
-    -fno-argument-alias
-    -fno-branch-count-reg
-    -fno-ipa-pure-const
-    -fno-ipa-reference
-    -fno-omit-frame-pointer
-    -fno-split-wide-types
-    -fno-tree-copy-prop
-    -fno-tree-dse
-    -fno-tree-sink
     )
 
 
Index: pcsx2.snapshot-3369/common/src/x86emitter/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3369.orig/common/src/x86emitter/CMakeLists.txt
+++ pcsx2.snapshot-3369/common/src/x86emitter/CMakeLists.txt
@@ -39,21 +39,9 @@
 
 # Remove optimization that can break the code. Must be retested
 set(BadFlags_O0
-    -fno-guess-branch-probability
-    -fno-dse
-    -fno-tree-dse
     )
 
 set(BadFlags_O1
-    -fno-argument-alias
-    -fno-branch-count-reg
-    -fno-ipa-pure-const
-    -fno-ipa-reference
-    -fno-omit-frame-pointer
-    -fno-split-wide-types
-    -fno-tree-copy-prop
-    -fno-tree-dse
-    -fno-tree-sink
     )
 
 
Index: pcsx2.snapshot-3369/pcsx2/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3369.orig/pcsx2/CMakeLists.txt
+++ pcsx2.snapshot-3369/pcsx2/CMakeLists.txt
@@ -36,21 +36,9 @@
 
 # Remove optimization that can break the code. Must be retested
 set(BadFlags_O0
-    -fno-guess-branch-probability
-    -fno-dse
-    -fno-tree-dse
     )
 
 set(BadFlags_O1
-    -fno-argument-alias
-    -fno-branch-count-reg
-    -fno-ipa-pure-const
-    -fno-ipa-reference
-    -fno-omit-frame-pointer
-    -fno-split-wide-types
-    -fno-tree-copy-prop
-    -fno-tree-dse
-    -fno-tree-sink
     )
 
 set(BadFlags_O2
