Already send upstream. Seem to cause some regression
Index: pcsx2.snapshot-3113/plugins/zzogl-pg/opengl/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3113.orig/plugins/zzogl-pg/opengl/CMakeLists.txt	2010-05-29 11:22:38.000000000 +0200
+++ pcsx2.snapshot-3113/plugins/zzogl-pg/opengl/CMakeLists.txt	2010-05-29 11:43:58.554840575 +0200
@@ -10,12 +10,6 @@
 set(CMAKE_C_FLAGS_RELEASE "")
 set(CMAKE_CXX_FLAGS_RELEASE "")
 
-# Also clear shared library flag to remove fpic option
-# that broke some asm code. The better solution will be to remove
-# the asm or fix it.
-set(CMAKE_SHARED_LIBRARY_CXX_FLAGS "")
-set(CMAKE_SHARED_LIBRARY_C_FLAGS "")
-
 # plugin name
 set(zzoglName zzogl)
 
@@ -29,6 +23,7 @@
     -msse2
     -fno-regmove
     -DZEROGS_SSE2
+    -fPIC
     )
 
 
