Index: pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/GSmain.cpp
===================================================================
--- pcsx2.snapshot-3208.orig/plugins/zzogl-pg/opengl/GSmain.cpp	2010-06-15 13:10:41.000000000 +0200
+++ pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/GSmain.cpp	2010-06-15 13:13:01.344197868 +0200
@@ -940,7 +940,7 @@
 	{
 		snapshotnr++;
 
-		sprintf(filename, "%ssnap%03ld.%s", path, snapshotnr, (conf.options&GSOPTION_TGASNAP) ? "bmp" : "jpg");
+		sprintf(filename, "%s/snap%03ld.%s", path, snapshotnr, (conf.options&GSOPTION_TGASNAP) ? "bmp" : "jpg");
 
 		bmpfile = fopen(filename, "rb");
 
