Index: pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/GSmain.cpp
===================================================================
--- pcsx2.snapshot-3208.orig/plugins/zzogl-pg/opengl/GSmain.cpp	2010-06-15 13:13:01.000000000 +0200
+++ pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/GSmain.cpp	2010-06-15 13:14:07.480949224 +0200
@@ -263,14 +263,11 @@
 
 	if (IsLogging())
 	{
+        fprintf(gsLog,"GS: ");
 		vfprintf(gsLog, fmt, list);
 		fprintf(gsLog, "\n");
 	}
 
-	printf("ZZogl-PG(GS): ");
-
-	vprintf(fmt, list);
-	printf("\n");
 	va_end(list);
 #endif
 }
@@ -284,15 +281,12 @@
 
 	if (IsLogging())
 	{
+        fprintf(gsLog,"WARNING: ");
 		vfprintf(gsLog, fmt, list);
 		fprintf(gsLog, "\n");
 	}
 
-	printf("ZZogl-PG(Warning): ");
-
-	vprintf(fmt, list);
 	va_end(list);
-	printf("\n");
 #endif
 }
 
@@ -305,17 +299,12 @@
 
 	if (IsLogging())
 	{
+        fprintf(gsLog,"DEBUG: ");
 		vfprintf(gsLog, fmt, list);
 		fprintf(gsLog, "\n");
 	}
 
-	printf("ZZogl-PG(Debug): ");
-
-	vprintf(fmt, list);
-	printf("\n");
 	va_end(list);
-
-
 #endif
 }
 
@@ -327,14 +316,10 @@
 
 	if (IsLogging())
 	{
+        fprintf(gsLog,"Error: ");
 		vfprintf(gsLog, fmt, list);
 		fprintf(gsLog, "\n");
 	}
-
-	printf("ZZogl-PG(Error): ");
-
-	vprintf(fmt, list);
-	printf("\n");
 	va_end(list);
 }
 };
Index: pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/ZZoglVB.cpp
===================================================================
--- pcsx2.snapshot-3208.orig/plugins/zzogl-pg/opengl/ZZoglVB.cpp	2010-06-15 13:10:41.000000000 +0200
+++ pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/ZZoglVB.cpp	2010-06-15 13:14:24.079949227 +0200
@@ -280,7 +280,7 @@
 
 	assert(pnewdepth != NULL && prndr != NULL);
 	if (pnewdepth->fbh != prndr->fbh) ZZLog::Debug_Log("pnewdepth->fbh(0x%x) != prndr->fbh(0x%x)", pnewdepth->fbh, prndr->fbh);
-	assert(pnewdepth->fbh == prndr->fbh);
+	// assert(pnewdepth->fbh == prndr->fbh);
 
 	if ((pprevdepth != pnewdepth) || (pprevdepth != NULL && (pprevdepth->status & CRenderTarget::TS_NeedUpdate)))
 		result = 2;
Index: pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/ZZoglFlush.cpp
===================================================================
--- pcsx2.snapshot-3208.orig/plugins/zzogl-pg/opengl/ZZoglFlush.cpp	2010-06-15 13:10:41.000000000 +0200
+++ pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/ZZoglFlush.cpp	2010-06-15 13:14:07.481948884 +0200
@@ -645,7 +645,7 @@
 	SET_STREAM();
 #ifdef _DEBUG
 	GLenum status = glCheckFramebufferStatusEXT(GL_FRAMEBUFFER_EXT);
-	assert(glCheckFramebufferStatusEXT(GL_FRAMEBUFFER_EXT) == GL_FRAMEBUFFER_COMPLETE_EXT);
+	// assert(glCheckFramebufferStatusEXT(GL_FRAMEBUFFER_EXT) == GL_FRAMEBUFFER_COMPLETE_EXT);
 #endif
 
 }
@@ -1867,7 +1867,7 @@
 
 	assert(curvb.prndr != NULL && curvb.pdepth != NULL);
 
-	assert(curvb.pdepth->fbh == curvb.prndr->fbh);
+	// assert(curvb.pdepth->fbh == curvb.prndr->fbh);
 
 	if (curvb.pdepth->status & CRenderTarget::TS_Virtual)
 	{
