Index: pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/GSmain.cpp
===================================================================
--- pcsx2.snapshot-3208.orig/plugins/zzogl-pg/opengl/GSmain.cpp	2010-06-15 13:14:07.480949224 +0200
+++ pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/GSmain.cpp	2010-06-15 13:14:34.157955576 +0200
@@ -530,7 +530,7 @@
 	SaveConfig();
 }
 
-void OnKeyboardF6(int shift)
+void OnKeyboardF7(int shift)
 {
 	FUNCLOG
 
@@ -556,7 +556,7 @@
 	SaveConfig();
 }
 
-void OnKeyboardF7(int shift)
+void OnKeyboardF6(int shift)
 {
 	FUNCLOG
 
Index: pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/CMakeLists.txt
===================================================================
--- pcsx2.snapshot-3208.orig/plugins/zzogl-pg/opengl/CMakeLists.txt	2010-06-15 13:15:07.402947013 +0200
+++ pcsx2.snapshot-3208/plugins/zzogl-pg/opengl/CMakeLists.txt	2010-06-15 13:15:18.296948496 +0200
@@ -39,7 +39,7 @@
 # Debug - Build
 if(CMAKE_BUILD_TYPE STREQUAL Debug)
     # add defines
-    add_definitions(${CommonFlags} -g -Wall -D_DEBUG)
+    add_definitions(${CommonFlags} -g -Wall -D_DEBUG -DZEROGS_DEVBUILD)
 endif(CMAKE_BUILD_TYPE STREQUAL Debug)
 
 # Devel - Build
